packages/api/tests/lambdas/test-sqs-message-consumer.js

```diff
@@ -362,3 +365,85 @@ test.serial('SQS message consumer queues correct number of workflows for rules m
     await cleanupRulesAndQueues(rules, [queue]);
   });
 });
+
+test.serial('processQueues archives messages from the ENABLED sqs rule only', async (t) => {
+  const { stackName } = process.env;
+  const { rules, queues } = await createRulesAndQueues();
+  const message = { testdata: randomString() };
+
+  // Send message to ENABLED queue
+  const firstMessage = await SQS.sendSQSMessage(
+    queues[0].queueUrl,
+    message
+  );
+
+  // Send message to DISABLED queue
+  const secondMessage = await SQS.sendSQSMessage(
+    queues[1].queueUrl,
+    { testdata: randomString() }
+  );
+  const enabledQueueKey = getS3KeyForArchivedMessage(stackName, firstMessage.MessageId);
+  const deadLetterKey = getS3KeyForArchivedMessage(stackName, secondMessage.MessageId);
+
+  await handler(event);
+
+  const item = await s3().getObject({
+    Bucket: process.env.system_bucket,
+    Key: enabledQueueKey,
+  }).promise();
+
+  t.deepEqual(message, JSON.parse(JSON.parse(item.Body.toString())));
+
+  t.false(s3ObjectExists({
+    Bucket: process.env.system_bucket,
+    Key: deadLetterKey,
+  }));
+
+  t.teardown(async () => {
+    await cleanupRulesAndQueues(rules, queues);
+  });
+});
+
+test.serial('processQueues archives multiple messages', async (t) => {
+  const { stackName } = process.env;
+  const { rules, queues } = await createRulesAndQueues();
+
+  // Send message to ENABLED queue
+  const messages = await Promise.all(
+    range(4).map(() =>
+      SQS.sendSQSMessage(
+        queues[0].queueUrl,
+        { testdata: randomString() }
+      ))
+  );
+  const deriveKey = (m) => getS3KeyForArchivedMessage(stackName, m.MessageId);
+  const keys = messages.map((m) => deriveKey(m));
+
+  // Send message to DISABLED queue
+  const secondMessage = await SQS.sendSQSMessage(
+    queues[1].queueUrl,
+    { testdata: randomString() }
+  );
+  const deadLetterKey = getS3KeyForArchivedMessage(stackName, secondMessage.MessageId);
+
+  await handler(event);
+
+  const items = await Promise.all(keys.map(async (k) =>
+    (s3().getObject({
+      Bucket: process.env.system_bucket,
+      Key: k,
+    })).promise()));
+
+  const msgBody = (m) => t.truthy(JSON.parse(JSON.parse(m.Body.toString())));
```

Comment:
If we're going to compare against every message, we might as well check that it has the exact expected content. Otherwise just doing a truthy comparison isn't very meaningful

Or we can just do an exists check here since the previous test verifies that the correct message is archived